pkg/zack/layout_split.go

```diff
@@ -0,0 +1,254 @@
+package zack
+
+import (
+	"github.com/charmbracelet/lipgloss"
+	"github.com/rs/zerolog/log"
+)
+
+type SplitOrientation uint8
+
+const (
+	Horizontal SplitOrientation = iota
+	Vertical
+)
+
+// the SplitLayoutModel should implement the Boundable interface
+// as it should resize the children based on the layout bounds
+var _ Boundable = (*SplitLayoutModel)(nil)
+
+// SplitLayoutModel is a layout that splits the screen into two views, either horizontally or vertically.
+// It takes a fraction (0 to 1) to determine how much space each view should take.
+// The left view will take the fraction of the screen, and the end view will take the remaining space.
+// For example, if the fraction is 0.4, the left view will take 40% of the screen and the end view will take 60%.
+//
+// Horizontal:
+//
+//	+------------------+------------------+
+//	| Start View (50%) | End View (50%)   |
```

Comment:
I don't know if this was on purpose but the example given above specifies 40% while this diagram shows a 50/50 split